pkg/windows/windows.go

```diff
@@ -191,12 +193,30 @@ func New(workerIgnitionEndpoint, vxlanPort string, instanceInfo *instance.Info,
 		return nil, errors.Wrapf(err, "unable to setup VM %s sshConnectivity", instanceInfo.Address)
 	}
 
+	// When in powershell, this will evaluate to an object containing information about the shell.
+	// When in cmd.exe, this will not evaluate to anything, and will be used literally. This can be used to determine
+	// if cmd or Powershell is being used.
+	powershellVersionTable := "($PSVersionTable)"
```

Comment:
Nice logic using `powershellVersionTable`.

Reply:
This looks fragile and more like a hack to me. Shouldn't you be querying whatever registry entry that defines the default shell?

Moreover do we really need to figure out what the default shell is? Can't we always launch a PowerShell given the that cmd commands will also work in PowerShell?

Reply:
Actually, looking at the registry adds a requirement that instances must be using OpenSSH as their SSH server... I'm not sure if we want to do that.